Mathlib/Data/Polynomial/Degree/Definitions.lean

```diff
@@ -140,18 +140,9 @@ theorem degree_eq_iff_natDegree_eq {p : R[X]} {n : ℕ} (hp : p ≠ 0) :
 
 theorem degree_eq_iff_natDegree_eq_of_pos {p : R[X]} {n : ℕ} (hn : 0 < n) :
     p.degree = n ↔ p.natDegree = n := by
-  constructor
-  · intro H
-    rwa [← degree_eq_iff_natDegree_eq]
-    rintro rfl
-    rw [degree_zero] at H
-    exact Option.noConfusion H
-  · intro H
-    rwa [degree_eq_iff_natDegree_eq]
-    rintro rfl
-    rw [natDegree_zero] at H
-    rw [H] at hn
-    exact lt_irrefl _ hn
+  obtain rfl|h := eq_or_ne p 0
```

Comment:
I'm not sure what the correct style is, but I've always been putting spaces around `|`.
```suggestion
  obtain rfl | h := eq_or_ne p 0
```

Reply:
Both styles occur.